scripts/kinit_helper

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+function kauth {
+    # If token doesn't exist, create it (will query for password)
+    if ! klist -s
+    then
+	while ! kinit -l 365d -r 365d
+	do
+            :
+	done
+    else
+	kinit -R &> /dev/null
+    fi
+    return 0
+}
+
+function afsauth {
+    # afs needs kerberos token
+    kauth
```

Comment:
I like this script, I'm probably going to be sourcing it and making use of both `kauth` and `afsauth` which are short, easy to remember function names.

Just one last tiny thing: the ordering of this step in `afsauth` and the ordering of the checks should probably be changed.

I think all the checks should happen prior to running `kauth` because `kauth` may prompt the user for their password. It's a bit annoying to type in your password, only afterwards to be told that you're on the wrong server.

I also think the group check should come before the hostname check, or both checks could be always be done. It'd just be nice for new team members to quickly be sent to the confluence page instead of sending them around to a different server first.

Reply:
Good idea. I switched the hostname and group checks and moved the kauth call so it would only be called if a token doesn't already exist.